src/components/PeopleList/PeopleList.tsx

```diff
@@ -0,0 +1,120 @@
+import { SortingFields } from '../../types/SortingFields';
+
+export const PeopleList = () => {
+  const { people, setWarning } = useContext(PeopleContext);
+  const [searchParams] = useSearchParams();
+  let filteredPeople = [...people];
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  if (!sort) {
+    filteredPeople = [...people];
+  }
+
+  if (sort === SortingFields.Name) {
+    if (!order) {
+      filteredPeople = filteredPeople.sort((a, b) =>
+        a.name.localeCompare(b.name),
+      );
+    } else {
+      filteredPeople = filteredPeople.sort((a, b) =>
+        b.name.localeCompare(a.name),
+      );
+    }
+  }
+
+  if (sort === SortingFields.Sex) {
+    if (!order) {
+      filteredPeople = filteredPeople.sort((a, b) =>
+        a.sex.localeCompare(b.sex),
+      );
+    } else {
+      filteredPeople = filteredPeople.sort((a, b) =>
+        b.sex.localeCompare(a.sex),
+      );
+    }
+  }
+
+  if (sort === SortingFields.Born) {
+    if (!order) {
+      filteredPeople = filteredPeople.sort((a, b) => a.born - b.born);
+    } else {
+      filteredPeople = filteredPeople.sort((a, b) => b.born - a.born);
+    }
+  }
+
+  if (sort === SortingFields.Died) {
+    if (!order) {
+      filteredPeople = filteredPeople.sort((a, b) => a.died - b.died);
+    } else {
+      filteredPeople = filteredPeople.sort((a, b) => b.died - a.died);
+    }
+  }
+
+  if (sex || query) {
+    if (sex) {
+      filteredPeople = filteredPeople.filter(
+        item => item.sex === searchParams.get(SortingFields.Sex),
+        // eslint-disable-next-line
+      ) ? filteredPeople.filter(// eslint-disable-next-line
+        item => item.sex === searchParams.get(SortingFields.Sex)) : people;
+    }
+
+    if (query) {
+      filteredPeople = filteredPeople.filter(item => {
+        if (
+          item.name.toLowerCase().includes(query.toLowerCase()) ||
+          (item.motherName &&
+            item.motherName.toLowerCase().includes(query.toLowerCase())) ||
+          (item.fatherName &&
+            item.fatherName.toLowerCase().includes(query.toLowerCase()))
+        ) {
+          return true;
+        }
+
+        return false;
+      });
+
+      if (!filteredPeople.length) {
+        setWarning('There are no people matching the current search criteria');
+      }
+    }
+  }
+
+  if (searchParams.get('centuries')) {
+    if (centuries) {
+      filteredPeople = filteredPeople.filter(item => {
+        if (!centuries.includes((Math.floor(item.born / 100) + 1).toString())) {
+          return false;
+        } else {
+          return true;
+        }
+      });
+    }
+  }
+
+  return (
+    <tbody>
+      {filteredPeople.map((person, _, array) => {
+        const motherLink = array.find(item => person.motherName === item.name);
```

Comment:
it's not a link yet
It's just object from array, so

```suggestion
        const mother = array.find(item => person.motherName === item.name);
```
same for prop below